packages/config-management/manage-generator/src/upgradeGenerator.ts

```diff
@@ -0,0 +1,45 @@
+import { WorkspaceConfigurationSchema } from "@fern-api/workspace-configuration";
+import { getGeneratorInvocationsByName } from "./generatorInvocations";
+
+export interface GeneratorUpgradeResult {
+    updatedWorkspaceConfiguration: WorkspaceConfigurationSchema;
+    upgrades: GeneratorUpgradeInfo[];
+}
+
+export interface GeneratorUpgradeInfo {
+    name: string;
+    previousVersion: string;
+    upgradedVersion: string;
+}
+
+export function upgradeGeneratorsIfPresent({
+    workspaceConfiguration,
+}: {
+    readonly workspaceConfiguration: WorkspaceConfigurationSchema;
+}): GeneratorUpgradeResult {
+    const generatorInvocationsByName = getGeneratorInvocationsByName();
+    const upgrades: GeneratorUpgradeInfo[] = [];
+    const updatedGenerators = workspaceConfiguration.generators.map((generatorConfig) => {
+        const updatedInvocation = generatorInvocationsByName[generatorConfig.name];
+        if (updatedInvocation != null && updatedInvocation.version !== generatorConfig.version) {
```

Comment:
nice thing about making the `Record` key stronger than `string` is that you can scrap this null check, and you'll get a compile error if you forget a generator in that map

Reply:
`updatedInvocation.version !== generatorConfig.version` should we make sure that the new version is later than the old version?